### ISSUE TITLE
[Bug]: Persistent Authentication Requests

### ISSUE THREAD

**Issue by Ate329** [denolehov/obsidian-git#721]
### Describe the bug

Upon starting Obsidian, an authentication window consistently appears, requesting GitHub credentials. This occurs even though I've previously cached GitHub credentials in Git using both the GitHub CLI and Git Credential Manager. I suspect that it is the option to "Pull updates on startup" that the plugin offers that seems to trigger this authentication prompt unnecessarily. Interestingly, despite not completing the authentication process, the plugin still functions properly. Additionally, closing the authentication window displays the following error:

![Error Image](https://github.com/denolehov/obsidian-git/assets/74974216/c8112532-5cce-4bca-8c1b-ec29ff174c9c)

The authentication window appears as follows:

![Authentication Window](https://github.com/denolehov/obsidian-git/assets/74974216/02308045-db3e-441e-9c1d-e71c2e248cc9)

This issue occurs on Fedora with KDE Plasma 5.

### Steps to reproduce

1. Start Obsidian.
2. Observe the authentication window appearing.

### Expected Behavior

Obsidian should start without prompting for GitHub credentials unnecessarily, especially if credentials have been cached.

### Addition context

- Operating System: Fedora
- Desktop Environment: KDE Plasma 5
- This is not a big issue but it disrupts workflow and presents a bit of inconvenience, particularly upon restarting Obsidian. Any insights or resolutions would be greatly appreciated.

### Operating system

Linux

### Installation Method

Flatpak

### Plugin version

2.24.1

**Comment by Vinzent03** [comment]
So your remote url uses the ssh protocol and it asks for your ssh key passphrase. Are you using the same remote and repository when working from the terminal with git or the github cli? Do you have to enter your password there as well every session, system start or never?

**Comment by Ate329** [comment]
Yes I am using the same remote and repository when using the terminal and when I try to push or pull in the terminal it works as normal. But I think I just found the issue. it might be caused by the fact that I am using GPG key to secure my Github credentials so I need to unlock the GPG key by entering my passphrase to allow the softwares to access it and somehow this pop-up window to enter my passphrase is not triggered by the software. Therefore, I don't think it's a issue caused by the plugin. Thanks for your time.

**Comment by JasonLovesDoggo** [comment]
For anyone who has the same problem in the future. My fix was to reclone my repo using the ssh git syntax (`git@github.com:user/repo.git`) vs the http one. (you could also change the remote url)

**Comment by muhmmdyssr** [comment]
> For anyone who has the same problem in the future. My fix was to reclone my repo using the ssh git syntax (`git@github.com:user/repo.git`) vs the http one. (you could also change the remote url)

Thanks a lot, it helped me solve the issue.
